server.go

```diff
@@ -226,9 +229,34 @@ func (a *App) handleScrubbing(ctx context.Context) error {
 	return nil
 }
 
+func (a *App) worker(ctx context.Context, in <-chan Envelope) {
+	for envelope := range in {
```

Comment:
you need also away to stop all workers (in case of shutdown) so
```go
for {
   var envelope Envelope
    select {
     case envelope =<-in:
     case <-ctx.Done():
         return
   }

  switch {...}
}
```